packages/components/src/components/common/switch/switch.tsx

```diff
@@ -20,7 +20,13 @@ const Switch = React.forwardRef<
       {...props}
       ref={ref}
     >
-      <SwitchPrimitives.Thumb className={cn(thumb)} />
+      {icon ? (
+        <SwitchPrimitives.Thumb className={cn(thumb)}>
+          <IvyIcon icon={icon.icon} rotate={icon.rotate} className={cn(thumbIcon({ size }))} />
+        </SwitchPrimitives.Thumb>
+      ) : (
+        <SwitchPrimitives.Thumb className={cn(thumb)} />
+      )}
```

Comment:
why duplicate the SwitchPrimitives.Thumb?
Should also work and is much smaller:
```tsx
<SwitchPrimitives.Thumb className={cn(thumb)}>
  {icon && <IvyIcon icon={icon.icon} rotate={icon.rotate} className={cn(thumbIcon({ size }))} />}
</SwitchPrimitives.Thumb>
```
Also you only need to use the cn utility if you want more than one class, but here `className={thumbIcon({size})}` should also do the job
and you don't need